docs-src/content/getting-started/streaming.md

```diff
@@ -101,11 +106,207 @@ audio = open('test.wav', 'rb')
 # print out results as they come in
 for resp in client.StreamingRecognize(cfg, audio):
     for result in resp.results:
-		if result.is_partial:
-			print("\r{0}".format(result.alternatives[0].transcript), end="")
-		else:
-			print("\r{0}".format(result.alternatives[0].transcript), end="\n")
+        if result.is_partial:
+            print("\r{0}".format(result.alternatives[0].transcript), end="")
+        else:
+            print("\r{0}".format(result.alternatives[0].transcript), end="\n")
+
+```
+{{% /tab %}}
+
+{{%/tabs %}}
+
+### Streaming from microphone
+
+The examples require an external library to talk to the sound driver,
+in order to stream audio from the microphone. Several libraries are
+available but the recommended ones are mentioned for each language below.
+
+{{%tabs %}}
+{{% tab "Go" %}}
+
+This example utilizes the [portaudio](https://github.com/gordonklaus/portaudio) bindings for Go
+to stream audio from a microphone. To use this package, [PortAudio](http://www.portaudio.com/)
+development headers and libraries to be installed. On Ubuntu this can be done using `apt-get install portaudio19-dev`.
+Note that this package is specific to linux distributions and can't be used on Windows or Mac.
+
+``` go
+package main
+
+import (
+    "bytes"
+    "context"
+    "encoding/binary"
+    "fmt"
+    "log"
+
+    "github.com/cobaltspeech/sdk-cubic/grpc/go-cubic"
+    "github.com/cobaltspeech/sdk-cubic/grpc/go-cubic/cubicpb"
+    "github.com/gordonklaus/portaudio"
+)
+
+const serverAddr = "127.0.0.1:2727"
+
+// Microphone object which can be passed to Cubic's streaming
+// recognition API as the audio stream. Two methods need to be
+// implemented for the Microphone object to : a Read method and
+// a Close method. (See below)
+type Microphone struct {
+    buffer []int16
+    stream *portaudio.Stream
+}
+
+// NewMicrophone instantiates a Microphone object with the desired
+// sampling rate and buffer size. When streaming to cubic, the sample
+// rate should be set to the sample rate of the model used. (See below)
+func NewMicrophone(sampleRate, bufferSize uint32) (*Microphone, error) {
+
+    mic := Microphone{buffer: make([]int16, bufferSize/2)} // each sample = 2 bytes
```

Comment:
Does `each sample = 2 bytes` depend on the sample rate?

Set a variable for `bufferSize/2` so you can use it here and on 167.

Reply:
No each sample just 16 bit = 2 byte. The cubic client measures it bufferSize in bytes while the PortAudio Microphone object deals in samples (be they 8bit or 16 bit etc.) So needed this conversion here to allow passing in of the same `bufferSize` used in `cubic.NewClient`